### PR TITLE
(feat) Re-add CICD and (to my great regret) the trunk checks.

### PR DESCRIPTION
**Pull request by rrw-zilliqa**
Update `passive-pruning.md` to explicitly mention that node operators should update their `zq2-*.toml` configuration files to include `do_checkpoints=true`. Add an example snippet showing the checkpoint parameters:

```toml
# Example checkpoint configuration
[load_checkpoint]
file = "checkpoint.zip"
hash = "abc123..."

do_checkpoints = true
``` 

This ensures compatibility with the ZIP64 checkpoint format introduced in v0.19.0-alpha.

### CODE REVIEW

File: .github/workflows/cicd-stg.yml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            governance-snapshot,
+            multisig,
+            neo-savant,
+            scilla-server,
+            zillion,
+            zilliqa-bridge-validator,
+            zilliqa-bridge-web,
+          ]
+        include:
+          - application: devex-apollo
```
Comment (frankmeds) [inline .github/workflows/cicd-stg.yml]: I am not sure we want to deploy all these app images :)

File: .github/workflows/cicd-prd.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    runs-on: ubuntu-22.04
+    name: "Build image with Makefile"
+    strategy:
+      fail-fast: false
+      matrix:
+        application: [developer-portal]
+        include:
+          - application: developer-portal
+            image_name: developer-portal
+            path: products/developer-portal
```
Comment (frankmeds) [inline .github/workflows/cicd-prd.yml]: this path looks not existing

File: .github/workflows/cicd-stg.yml [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    branches:
+      - main
+      - users/richard/readd-cicd
+  # On merged
+  push:
+    branches:
+      - main
+
+jobs:
+  build-docker:
```
Comment (frankmeds) [inline .github/workflows/cicd-stg.yml]: this job looks not required

File: .github/workflows/cicd-stg.yml (hunk 1 above)
Comment (rrw-zilliqa) [reply .github/workflows/cicd-stg.yml]: Good point well made - sorry! - removed.

File: .github/workflows/cicd-prd.yml (hunk 2 above)
Comment (rrw-zilliqa) [reply .github/workflows/cicd-prd.yml]: Good point well made ..

**Review by frankmeds — APPROVED** [review]

**Pull request merged**